### PR TITLE
Fix label ordering in Visualizer

### LINKED ISSUE #225: vis.visualize_dataset() labels are off (by one?)
I ran some of the README.md example code using the KITTI dataset and noticed that the labels are wrong.  I noticed this too with my custom dataset.

```
import open3d.ml as _ml3d
import open3d.ml.torch as ml3d # or open3d.ml.tf as ml3d

dataset = ml3d.datasets.SemanticKITTI(dataset_path='/media/equant/7fe7f0a0-e17f-46d2-82d3-e7a8c25200bb/work/SemanticKITTI/',
    test_split=[ '12' ],
    training_split=[ '10' ],
    validation_split=['08'],
    all_split=[ '08', '10', '12' ],
)   

# get the 'all' split that combines training, validation and test set
all_split = dataset.get_split('all')

# print the attributes of the first datum
print(all_split.get_attr(0))

# print the shape of the first point cloud
print(all_split.get_data(0)['point'].shape)

# show the first 100 frames using the visualizer
vis = ml3d.vis.Visualizer()
vis.visualize_dataset(dataset, 'all', indices=range(1))
```

With a little inspection I hope you can see that the labels reported are not correct.
The points with the label "traffic-sign" correspond to what should be vegetation.  What is labeled "trunk" are points that should be labeled "unlabeled".   Etc.

Screenshot_2021-02-17_10-17-03

### LINKED ISSUE #243: Visualizer with LabelLUT prints wrong labels when one of the classes is not present in the data.
I've encountered a peculiar bug: when visualising a pointcloud using `vis.visualize(data)` with a pre-defined LabelLUT, the visualisation goed wrong when one of the classes is not present in the data.

Example: I ran **RandLA-Net** with pre-trained weights on the **Paris-Lille-3D validation set** (using the code snippet from the readme). It outputs predictions for the classes [1, ..., 9], where 0 is ignored ('unknown class'). Therefore, the data has no points labelled as 0. When I initialise the visualizer with the full set of labels, including class '0', all labels <=4 are shifted 1 down, class 4 is 'missing', and all other classes are shown correctly. When I initialise the visualiser without class 0, it shows everything correctly.

When I set all labels for class '1' to '0' (so now the data does not contain any point of class '1', the same happens: classes 2, 3, and 4 are shifted down, and 4 disappears.

Code:
```
result = pipeline.run_inference(data)

dataset = ml3d.datasets.ParisLille3D(data_dir)
test_split = dataset.get_split("val")
data = test_split.get_data(0)

paris_labels = ml3d.datasets.ParisLille3D.get_label_to_names()
# paris_labels.pop(0)
v = ml3d.vis.Visualizer()
lut = ml3d.vis.LabelLUT()
for val in sorted(paris_labels.keys()):
    lut.add_label(paris_labels[val], val)
v.set_lut("labels", lut)

data_viz = {}
data_viz['name'] = 'randla_npm3d_val'
data_viz['points'] = data['point']
data_viz['labels'] = result['predict_labels'] + 1

v.visualize([data_viz])
```

Example: here the visualisation is correct for class 4.
Example of correct behaviour

The same data, but now the LabelLUT includes class 0. The data for class 4 is now shown as class 3. Nothing is shown for class 4. Classes >4 are shown correctly (excluded from screenshot for clarity).
Example of incorrect behaviour

A quick fix is to set any single label to the missing class (e.g.) 0:
```
data_viz['labels'][0] = 0
```

### PR DESCRIPTION
**Pull request by sanskar107**
Fix wrong label order in visualizer when one of the classes is not present.
Fixes #225 #243 #409

---
This change is [Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/422)

### CODE REVIEW

**Review by errissa — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/isl-org/open3d-ml/422)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @sanskar107)
